lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4004,15 +4014,32 @@ namespace {
       if (funcTemplate) {
         unsigned i = 0;
         for (auto param : *funcTemplate->getTemplateParameters()) {
+          auto templateTypeParam = cast<clang::TemplateTypeParmDecl>(param);
+          // If the template type parameter isn't used in the signature then we
```

Comment:
Actually this makes me wonder, what happens with template functions who don't use the templated parameters at all? Would something like `template <typename T> void f() {}` be a different patch entirely or could we just pick some primitive type to use as a specialization?

Reply:
I think picking a random primitive would lead to users incorrectly calling `f()` in the worst case and in the best case just wouldn't be that helpful. If there's a template type param, I hope it's there for a reason. 

To support these kinds of templates, I think we talked about importing them like this:
```
func f<T>(_ _: T.Type)
```

But that will need an evolution discussion.